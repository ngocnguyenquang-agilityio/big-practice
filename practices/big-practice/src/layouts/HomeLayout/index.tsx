import Collection from "@components/Collection"
import { Outlet } from "react-router-dom"

const mockListCollection = ['All', 'Hoodie', 'Jacket', 'Shirt']

const HomeLayout = () => {
  return (
    <>
      <div className="order-first w-full flex-none md:max-w-[125px]">
        <Collection list={mockListCollection} title="Collection" />
      </div>
      <Outlet />
      <div className="order-none flex-none md:order-last md:w-[125px]">
        <Collection list={mockListCollection} title="Sort by" />
      </div>
    </>
  )
}

export default HomeLayout
