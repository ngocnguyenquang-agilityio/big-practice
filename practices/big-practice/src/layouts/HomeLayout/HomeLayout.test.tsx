import { RenderResult, render } from '@testing-library/react';

// Components
import HomeLayout from '.';

const MockLeftAside = () => <div>Left Aside</div>;
const MockRightAside = () => <div>Right Aside</div>;

const props = {
  leftAside: <MockLeftAside />,
  rightAside: <MockRightAside />,
};

describe('HomeLayout', () => {
  let comp: RenderResult;

  beforeEach(() => {
    comp = render(
      <HomeLayout {...props}>
        <div>Children</div>
      </HomeLayout>,
    );
  });

  test('Should render correctly', () => {
    expect(comp).toBeTruthy();
  });

  test('Should render correctly', () => {
    expect(comp).toMatchSnapshot();
  });
});
