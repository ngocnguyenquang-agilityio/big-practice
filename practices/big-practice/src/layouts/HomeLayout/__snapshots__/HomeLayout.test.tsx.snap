// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeLayout Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="order-first w-full flex-none md:max-w-[125px]"
      >
        <div>
          Left Aside
        </div>
      </div>
      <div
        class="order-last min-h-screen w-full md:order-none"
      >
        <div>
          Children
        </div>
      </div>
      <div
        class="order-none flex-none md:order-last md:w-[125px]"
      >
        <div>
          Right Aside
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="order-first w-full flex-none md:max-w-[125px]"
    >
      <div>
        Left Aside
      </div>
    </div>
    <div
      class="order-last min-h-screen w-full md:order-none"
    >
      <div>
        Children
      </div>
    </div>
    <div
      class="order-none flex-none md:order-last md:w-[125px]"
    >
      <div>
        Right Aside
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
